datalad/metadata/indexers/base.py

```diff
@@ -0,0 +1,46 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+""" Metadata indexer base class """
+
+from typing import Any, Dict, List, Union
+
+
+class MetadataIndexer(object):
```

Comment:
added a comment in the test that likely MockedIndexer should subclass this one.

Since it defines interface, should it may be use `abc` module and define `@abstractmethod`s so subclasses then are guaranteed to implement this interface "fully"?

Reply:
Good idea. Done.